api/src/core/v1.rs

```diff
@@ -282,7 +282,8 @@ pub struct NodeSelector {
 #[derive(Serialize, Deserialize, Default, Debug, Clone, PartialEq)]
 #[serde(rename_all = "camelCase")]
 pub struct NodeSelectorTerm {
-    pub match_expressions: Vec<NodeSelectorRequirement>,
+    pub match_expressions: Option<Vec<NodeSelectorRequirement>>,
```

Comment:
```rust
#[serde(default)]
pub foo Vec<T>,
```

(ie: a list that defaults to empty) is much easier to use than an `Option<Vec<T>>`, especially since `None` and `[]` are semantically equivalent (in almost all cases in k8s API).

(ditto next line)

Reply:
done